File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxx = INT_MIN, n = nums.size();
        int sum = 0;
        for(int i = 0; i < n; i++) {
            sum += nums[i];
            maxx = max(maxx, sum);
            if(sum < 0) {
                sum = 0;
            }
        }
        return maxx;
    }
};

File: 53-maximum-subarray/maximum-subarray.go
func maxSubArray(nums []int) int {
    maxx := -1000000
    sum := 0
    for i := 0; i < len(nums); i++{
        sum += nums[i]
        if sum > maxx{
            maxx = sum
        }
        if sum < 0{
            sum = 0
        }
    }
    return maxx
}